oauthlib/oauth2/rfc6749/grant_types/refresh_token.py

```diff
@@ -101,6 +101,8 @@ def validate_token_request(self, request):
             if not self.request_validator.authenticate_client(request):
                 log.debug('Invalid client (%r), denying access.', request)
                 raise errors.InvalidClientError(request=request)
+            # Ensure that request.client_id is set.
+            request.client_id = request.client_id or request.client.client_id
```

Comment:
can you please add some test to verify the change and it won't introduce any regression?

Reply:
Also, please rebase because `lint_python` should now pass.

Reply:
Okay, added a test and rebased.